dces-drc-integration/src/main/java/uk/gov/justice/laa/crime/dces/integration/rest/handlers/DcesDrcErrorController.java

```diff
@@ -0,0 +1,52 @@
+package uk.gov.justice.laa.crime.dces.integration.rest.handlers;
+
+import lombok.AllArgsConstructor;
+import lombok.extern.slf4j.Slf4j;
+import org.apache.commons.lang3.StringUtils;
+import org.springframework.boot.web.error.ErrorAttributeOptions;
+import org.springframework.boot.web.servlet.error.DefaultErrorAttributes;
+import org.springframework.boot.web.servlet.error.ErrorAttributes;
+import org.springframework.boot.web.servlet.error.ErrorController;
+import org.springframework.http.HttpStatus;
+import org.springframework.web.bind.annotation.RequestMapping;
+import org.springframework.web.bind.annotation.RestController;
+import org.springframework.web.context.request.WebRequest;
+import uk.gov.justice.laa.crime.dces.integration.rest.common.ErrorResponse;
+import uk.gov.justice.laa.crime.dces.integration.tracing.TraceService;
+
+import java.util.Map;
+
+
+@Slf4j
+@RestController
+@AllArgsConstructor
+public class DcesDrcErrorController implements ErrorController {
+
+    private final TraceService traceService;
+    public static final String MESSAGE_KEY = "message";
+    public static final String STATUS_KEY = "status";
+    public static final String ERROR_KEY = "error";
+    public static final String PATH_KEY = "path";
+
+    private final ErrorAttributes errorAttributes = new DefaultErrorAttributes();
+
+    @RequestMapping("/error")
+    public ErrorResponse handleError(final WebRequest webRequest, final Exception exception) {
+        log.error("Error occurred while processing web request ", exception);
+        final Map<String, Object> errorAttributesMap = this.errorAttributes.getErrorAttributes(webRequest, ErrorAttributeOptions.defaults());
+        return createErrorResponse(errorAttributesMap);
+    }
+
+    private ErrorResponse createErrorResponse(final Map<String, Object> errorAttributes) {
+        final int statusCode = (Integer) errorAttributes.getOrDefault(STATUS_KEY, HttpStatus.INTERNAL_SERVER_ERROR.value());
+        final String errorMessage = errorAttributes.getOrDefault(MESSAGE_KEY, StringUtils.EMPTY).toString();
+        final String errorCode = errorAttributes.getOrDefault(ERROR_KEY, StringUtils.EMPTY).toString();
```

Comment:
What I meant, was the getOrDefault is un-needed. As it is already defaulted.
The code, at first glance, should set the statuscode to an empty string. i.e. there's no need to null check, or check for default. It will always be populated. So a errorAttributes.get(ERROR_KEY)

The code that sets the defaults has the following:
		if (status == null) {
			errorAttributes.put("status", 999);
			errorAttributes.put("error", "None");
			return;
So the orDefault is un-needed.